packages/autocomplete-core/src/onInput.ts

```diff
@@ -55,21 +55,6 @@ export function onInput<TItem>({
   setHighlightedIndex(props.defaultHighlightedIndex);
   setQuery(query);
 
-  if (query.length < props.minLength) {
```

Comment:
We want to keep this branch, but the condition should be `query.length === 0 && props.openOnFocus === false`. This makes sure to "reset" the state.

The reasoning behind this is that as opposed to InstantSearch, when there's no query, we do not want to consider any results, _except_ when `openOnFocus` is `true`. This behavior would be easier to validate once we write integration tests.